composer.json

```diff
@@ -1,15 +1,46 @@
 {
-	"name": "silverstripe/redirectedurls",
-	"description": "Provides a system for users to configure arbitrary redirections in the CMS",
-	"license": "BSD-3-Clause",
-	"type": "silverstripe-module",
-	"keywords": ["silverstripe", "urls", "redirectedurls", "redirect"],
-	"authors": [{
-		"name": "Sam Minnee",
-		"email": "sam@silverstripe.com"
-	}],
-	"require": {
-		"silverstripe/framework": "~3.1",
-        "silverstripe/cms": "~3.1"
-	}
+    "name": "silverstripe/redirectedurls",
+    "description": "Provides a system for users to configure arbitrary redirections in the CMS",
+    "license": "BSD-3-Clause",
+    "type": "silverstripe-vendormodule",
+    "keywords": [
+        "silverstripe",
+        "urls",
+        "redirectedurls",
+        "redirect"
+    ],
+    "authors": [
+        {
+            "name": "Sam Minnee",
+            "email": "sam@silverstripe.com"
+        }
+    ],
+    "require": {
+        "silverstripe/framework": "^4",
+        "silverstripe/cms": "^4"
+    },
+    "require-dev": {
+        "phpunit/phpunit": "^5",
+        "squizlabs/php_codesniffer": "^3"
+    },
+    "autoload": {
+        "psr-4": {
+            "SilverStripe\\RedirectedURLs\\": "src/"
+        }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "SilverStripe\\RedirectedURLs\\Test\\": "tests/"
```

Comment:
We should put the tests autoloader definition into `autoload` rather than `autoload-dev`

Reply:
How come? Hopefully our non-dev zips won't have the tests folder and it may cause a composer error on production sites if we have the autoloader defined for non existing directories...

Reply:
Just had to clarify this before I replied - the autoload-dev separation is only valid for [root packages](https://getcomposer.org/doc/04-schema.md#root-package), not dependency packages, so by implementing this here the autoloader wouldn't register test classes unless the module is your root project

Reply:
ah - hmm... I guess that's a problem if your project relies on the test classes somehow...

ok - we'll just have to be careful that this doesn't actually break anything with the dist files

Reply:
@andrewandante has done this now :)